agent/consul/state/acl_oss.go

```diff
@@ -289,11 +289,11 @@ func (s *Store) aclTokenListGlobal(tx *txn, _ *structs.EnterpriseMeta) (memdb.Re
 	return tx.Get("acl-tokens", "local", false)
 }
 
-func (s *Store) aclTokenListByPolicy(tx *txn, policy string, _ *structs.EnterpriseMeta) (memdb.ResultIterator, error) {
+func aclTokenListByPolicy(tx ReadTxn, policy string, _ *structs.EnterpriseMeta) (memdb.ResultIterator, error) {
```

Comment:
There are a bunch of `Store` methods that do not actually use the `Store`. By making these functions instead of methods we remove the need to pass around a `Store`.

I could make this change more generally and remove all of the method receivers, but for now I only changed the few that are called from `ProcessChanges`.

The `ReadTxn` will let us call these with either a wrapped or unwrapped txn. Since the wrapping doesn't do anything special for reads. This change was only necessary before d49d67ee70151bcd7037b1bc62215d4fddc3d7b7, but I've left it for now in case that commit is not viable.

Generally using an interface here seems like it has some nice properties, now that we have more than one possible implementation of a txn. If we took this further it would allow us to move EventPublisher out of `state` and into the `stream` package, so that all of the streaming logic is together in a single package.

Reply:
PR to convert all of these: https://github.com/hashicorp/consul/pull/8282